vm/actor/src/builtin/market/mod.rs

```diff
@@ -390,46 +491,57 @@ impl Actor {
         let miner_addr = *rt.message().caller();
 
         rt.transaction::<State, Result<(), ActorError>, _>(|st, rt| {
-            let prop = Amt::load(&st.proposals, rt.store())
-                .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
-            let mut states = Amt::load(&st.states, rt.store())
-                .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+            let mut msm = st.mutator(rt.store());
+            msm.with_deal_states(Permission::Write)
+                .with_deal_proposals(Permission::ReadOnly)
+                .build()
+                .map_err(|e| actor_error!(ErrIllegalState; "failed to load state: {}", e))?;
 
             for id in params.deal_ids {
-                let deal: DealProposal = prop
-                    .get(id)
-                    .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?
-                    .ok_or_else(|| {
-                        ActorError::new(
-                            ExitCode::ErrIllegalState,
-                            "Failed to retrieve DealProposal".to_owned(),
-                        )
-                    })?;
-                assert_eq!(deal.provider, miner_addr);
+                let deal = msm.deal_proposals.as_ref().unwrap().get(id).map_err(
+                    |e| actor_error!(ErrIllegalState; "failed to get deal proposal: {}", e),
+                )?;
+                // deal could have terminated and hence deleted before the sector is terminated.
+                // we should simply continue instead of aborting execution here if a deal is not found.
+                if deal.is_none() {
+                    continue;
+                }
+                let deal = deal.unwrap();
+
+                assert_eq!(
```

Comment:
I see they do an assert, in specs-actor, but do we actually wanna kill the program?

Reply:
ah yeah, didn't think too much in detail about this one, seems very weird that they are asserting here (I will flag to them)